app/views/films/show.html.haml

```diff
@@ -3,7 +3,11 @@
 - if @film.film_link.attached?
   = video_tag rails_blob_path(@film.film_link), controls: true, width: '100%'
 %div
-  = link_to t('destroy'), @film,
-  method: :delete, data: { confirm: t('are_you_sure') }, class: 'btn btn-danger'
+  - if current_user.admin?
+    - if @film.film_link.attached?
```

Comment:
also business logic in views isn't cool. You can try to move it to other place. Your film is a object, so maybe to decorators? https://github.com/drapergem/draper